app/controllers/admin/whitelists_controller.rb

```diff
@@ -1,3 +1,52 @@
 class Admin::WhitelistsController < AdminController
-  def new; end
+  def new
+    @whitelist ||= Whitelist.new
+
+    if organisation_name_submitted? && organisation_name_needs_validation?
+      @organisation_name = CustomOrganisationName.new(name: whitelist_params.dig(:organisation_name))
+      if @organisation_name.invalid?
+        params[:whitelist][:step] = "fourth"
```

Comment:
You could refactor this to be a guard statement:
```
      return params[:whitelist][:step] = "fourth" if @organisation_name.invalid?

      params[:whitelist][:organisation_name_valid] = true
    end
```

Reply:
or a ternary? e.g.
`@organisation_name.invalid? ? params[:whitelist][:step] = "fourth" : params[:whitelist][:organisation_name_valid]`